compiler/pash_runtime_daemon.py

```diff
@@ -262,10 +270,53 @@ def run(self, in_filename, out_filename):
                 self.fout = open(out_filename, "w")
 
             self.parse_and_run_cmd(input_cmd)
-            
+        
+        self.reader.close()
         shutdown()
 
 
+class UnixPipeReader:
+    def __init__(self, in_filename, blocking = True):
+        self.in_filename = in_filename
+        self.buffer = ""
+        self.blocking = blocking
+        if not self.blocking:
+            self.fin = open(self.in_filename)
+
+    def get_next_cmd(self):
```

Comment:
We should add a comment explaining what this method does. What can the possible starting states be (e.g., buffer can contain complete commands, at least one, no more than X, etc). And then what happens when the get_next_cmd is done. What is returned and what should we expect from the state.